PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m

```diff
@@ -89,156 +92,123 @@ + (NSArray *) load_data_from_keychain {
+        for (cookie in [cookieJar cookies]) {
+            NSLog(@"%@", cookie);
+        }
+        
+        NSDictionary * headers = [NSHTTPCookie requestHeaderFieldsWithCookies:[cookieJar cookies]];
+        [request setAllHTTPHeaderFields:headers];
+        
+    }
+    
+    NSLog(@"Request %@", request);
     
-    connectionStatus = 0;
+    communicationStatus = 1;
     
-    if (!error) {
+    if (body != NULL) {
+        
+        NSError *error = nil;
+        NSData *data = [NSJSONSerialization dataWithJSONObject:body
+                                                       options:kNilOptions error:&error];
+        
+        if (!error) {
+            NSURLSessionUploadTask *uploadTask = [session uploadTaskWithRequest:request
+                                                                       fromData:data completionHandler:^(NSData *data,NSURLResponse *response,NSError *error) {
+                                                                           NSString *datastring = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+                                                                           NSLog(@"Data %@", datastring);
+                                                                           NSLog(@"Response %@", response);
+                                                                           if (error != NULL) {
+                                                                               NSLog(@"Error %@", error);
+                                                                           } else {
+                                                                               communicationStatus = 0;
+                                                                           }
+                                                                           
+                                                                       }];
+            
+            [uploadTask resume];
+        }
+        
+        sleep(1);
+        
+    } else {
+        
         NSURLSessionDataTask *uploadTask = [session dataTaskWithRequest:request completionHandler:^(NSData *data,NSURLResponse *response,NSError *error) {
             NSString *datastring = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
             NSLog(@"Data %@", datastring);
             NSLog(@"Response %@", response);
             if (error != NULL) {
                 NSLog(@"Error %@", error);
-                
             }
             
-            NSString *compareThisPartOfTheString = [datastring substringToIndex:14];
-            
-            if ([compareThisPartOfTheString isEqual: @"{\"status\":\"ok\""]) {
-                
-                connectionStatus = 1;
+            if (test_connection == 1) {
+                NSString *compareThisPartOfTheString = [datastring substringToIndex:14];
                 
+                if ([compareThisPartOfTheString isEqual: @"{\"status\":\"ok\""]) {
```

Comment:
yuk yuk yuk. Please deserialize the json properly (but maybe you did this in the later PR?)

Reply:
this has been done